File: 2021/10/1.py
char_map = {
    '(': ')',
    '[': ']',
    '{': '}',
    '<': '>',
}

illegal_chars = []

with open("input.txt") as f:
    for line in f:
        left = ""
        right = ""
        for char in line.rstrip():
            if char == '(' or char == '[' or char == '{' or char == '<':
                left += char
            else:
                right += char
                if char != char_map[left[-1]]:
                    illegal_chars.append(char)
                    break
                else:
                    left = left[:-1]

error_score = 0
for char in illegal_chars:
    if char == ')':
        error_score += 3
    elif char == ']':
        error_score += 57
    elif char == '}':
        error_score += 1197
    elif char == '>':
        error_score += 25137

print(error_score)

File: 2022/3/1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"unicode"
)

func main() {
	f, _ := os.Open("input.txt")

	defer f.Close()

	scanner := bufio.NewScanner(f)
	var totalPriority int32 = 0

	for scanner.Scan() {
		rucksack := scanner.Text()
		rucksackLength := len(rucksack)
		compartment1 := rucksack[:rucksackLength/2]
		compartment2 := rucksack[rucksackLength/2:]

		charMap := map[string]bool{}
		for _, charInt := range compartment1 {
			charMap[string(charInt)] = true
		}

		for _, charInt := range compartment2 {
			char := string(charInt)
			if charMap[char] {
				if unicode.IsUpper(charInt) {
					totalPriority += charInt - int32(38)
				} else {
					totalPriority += charInt - int32(96)
				}
				break
			}
		}
	}
	fmt.Printf("Total priority is: %d", totalPriority)
}
